compiler/ir/backend.jvm/lower/src/org/jetbrains/kotlin/backend/jvm/lower/FragmentLocalFunctionPatchLowering.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2010-2022 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.backend.jvm.lower
+
+import org.jetbrains.kotlin.backend.common.FileLoweringPass
+import org.jetbrains.kotlin.backend.common.IrElementTransformerVoidWithContext
+import org.jetbrains.kotlin.backend.common.lower.LocalDeclarationsLowering
+import org.jetbrains.kotlin.backend.common.phaser.makeIrFilePhase
+import org.jetbrains.kotlin.backend.jvm.JvmBackendContext
+import org.jetbrains.kotlin.backend.jvm.ir.createJvmIrBuilder
+import org.jetbrains.kotlin.backend.jvm.localDeclarationsPhase
+import org.jetbrains.kotlin.backend.jvm.lower.FragmentSharedVariablesLowering.Companion.GENERATED_FUNCTION_NAME
+import org.jetbrains.kotlin.ir.IrStatement
+import org.jetbrains.kotlin.ir.ObsoleteDescriptorBasedAPI
+import org.jetbrains.kotlin.ir.builders.declarations.addValueParameter
+import org.jetbrains.kotlin.ir.builders.irCall
+import org.jetbrains.kotlin.ir.builders.irGet
+import org.jetbrains.kotlin.ir.declarations.*
+import org.jetbrains.kotlin.ir.expressions.IrCall
+import org.jetbrains.kotlin.ir.expressions.IrExpression
+import org.jetbrains.kotlin.ir.expressions.copyTypeArgumentsFrom
+import org.jetbrains.kotlin.ir.visitors.transformChildrenVoid
+
+
+// Used from CodeFragmentCompiler for IDE Debugger Plug-In
+@Suppress("unused")
+val fragmentLocalFunctionPatchLowering = makeIrFilePhase(
+    ::FragmentLocalFunctionPatchLowering,
+    name = "FragmentLocalFunctionPatching",
+    description = "Rewrite calls to local functions to the appropriate, lifted function created by local declarations lowering.",
+    prerequisite = setOf(localDeclarationsPhase)
+)
+
+// This lowering rewrites local function calls in code fragments to the
+// corresponding lifted declaration. In the process, the lowering determines
+// whether the captures of the local function are a subset of the captures of
+// the fragment, and if not, introduces additional captures to the fragment
+// wrapper. The captures are then supplied to the fragment wrapper as
+// parameters supplied at evaluation time.
+internal class FragmentLocalFunctionPatchLowering(
+    val context: JvmBackendContext
+) : IrElementTransformerVoidWithContext(), FileLoweringPass {
+
+    override fun lower(irFile: IrFile) {
+        if (irFile.name != "fragment.kt") return
```

Comment:
Seems ad-hoc and kind of dangerous. :) Why is it needed?

Reply:
You're absolutely right, must have been an ad-hoc and kinda dangerous duct-tape solution to some intermediate state of the code :D 

I have no record of why I introduced it, and `visitSimpleFunction` right after is just as defensive.

Out it goes!